data_analysis.py

```diff
@@ -0,0 +1,81 @@
+import pandas as pd
+from openai import OpenAI
+
+STOCK_LIST =  ['NVDA', 'TSLA', 'AMZN', 'MSFT', 'COST']
+BASE_PATH = '/Users/neo_chara/Desktop/Projects/Stock-Watch'
+
+def get_clean_data():
+    stocks_dfs_dict = {}
+    for symbol in STOCK_LIST:
+        file_name = f"{symbol.lower()}_stock.csv"
+        path_csv = f"{BASE_PATH}/data/{file_name}"
+        stocks_dfs_dict[f"{symbol.lower()}"] = pd.read_csv(path_csv,index_col='Date')
+
+    return stocks_dfs_dict
+
+
+def get_description():
+    stocks_dict = get_clean_data()
+
+    description_dict = {}
+    for stock_idx, stock_data in stocks_dict.items():
+        description_dict[stock_idx] = stock_data.describe()
+    
+    return description_dict
+
+def _filter_description(descr_dict):
+    """Filters description dict to keep indices of interest and Close column"""
+    indices_to_keep = ["mean", "min", "max", "std"]
+    characteristic_to_keep = "Close"
+
+    #restricted_data = {key: df.loc[indices_to_keep] for key, df in descr_dict.items()}
+    restricted_dict = {}
+    for key, df in descr_dict.items():
+        restricted_dict[key] = df[characteristic_to_keep].loc[indices_to_keep]
+    
+    return restricted_dict
+
+
+def _get_llm_response(idx_queried, idx_descr_df):
+    client = OpenAI()
+    df_str = idx_descr_df.to_csv(index=True)
```

Comment:
Better if LLM response doesn't also edit the data. Rather just pass the correct data format directly

Reply:
ok